scripts/test-truncating-of-client-hello.py

```diff
@@ -136,66 +133,86 @@ def main():
                                 ("hello truncate", -13, 0),
                                 ]:
 
-        conversation = Connect(host, port)
-        node = conversation
+        conversation_name = None
+        if "pad" in name:
+            conversation_name = name + ": " + str(pad_len) + " of \"" + str(pad_byte) + "\" byte padding"
+        else:
+            conversation_name = name + ": " + str(-pad_len) + " of bytes truncated"
+        conversation = Conversation(conversation_name)
+
+        conversation.root_node = Connect(host, port)
         ciphers = [CipherSuite.TLS_RSA_WITH_AES_128_CBC_SHA]
-        node = node.add_child(pad_handshake(ClientHelloGenerator(ciphers,
+        node = conversation.root_node.add_child(pad_handshake(ClientHelloGenerator(ciphers,
                                                    extensions={ExtensionType.renegotiation_info: None}),
                                             pad_len, pad_byte))
         # we expect Alert and Close or just Close
         node = node.add_child(ExpectAlert())
         node.next_sibling = ExpectClose()
         node.add_child(ExpectClose())
+        conversations.append(conversation)
 
-        if "pad" in name:
-            conversations[name + ": " + str(pad_len) + " of \"" + str(pad_byte) +
-                          "\" byte padding"] = conversation
+    # first and last test is sanity
+    conversations.insert(0, conversation_sanity)
+    conversations.append(conversation_sanity)
+
+    # TBD: move it to separate function
+    for conversation in conversations:
+        if run_only and conversation.name not in run_only or conversation.name in run_exlude:
+            conversations.remove(conversation)
+
+    run_all(conversations)
+
+def all_passed(conversations):
+    for conversation in conversations:
+        if not conversation.passed:
+            return False
+
+    return True
+
+def print_result(conversations):
+    print("List of Succesfull Tests:")
+    for conversation in conversations:
```

Comment:
I wanted improve readability but I think problem was in way of presenting results as you mentioned (3 lines for every conversation). I have returned previous sorting and updated printing fn and I think it looks better now.

I am not sure about verbose. There are several ways. Would you like to use some simple function with verbose flag which could be set by user as user input or some different approach?